neoexchange/photometrics/gf_movie.py

```diff
@@ -269,14 +236,48 @@ def update(n):
         except InvalidTransformError:
             pass
         # finish up plot
-        current_count = len(np.unique(fits_files[:n+1]))
-        ax.set_title('UT Date: {} ({} of {})'.format(date.strftime('%x %X'), current_count, int(len(fits_files)-(copies-1)*start_frames)), pad=10)
+        current_count = len(np.unique(fits_files[:n + 1]))
+        ax.set_title('UT Date: {} ({} of {})'.format(date.strftime('%x %X'), current_count,
+                                                     int(len(fits_files) - (copies - 1) * start_frames)), pad=10)
+
+        # Set frame to be center of chip in arcmin
+        shape = data.shape
+        x_frac = 0
+        y_frac = 0
+        if center is not None:
+            y_frac = np.max(int((shape[0] - (center * 60)/header_n['PIXSCALE']) / 2), 0)
+            x_frac = np.max(int((shape[1] - (center * 60)/header_n['PIXSCALE']) / 2), 0)
+
+            # set data ranges
+            data_x_range = [x_frac, -(x_frac+1)]
+            data_y_range = [y_frac, -(y_frac+1)]
+            if target_data:
+                nonlocal x_offset
+                nonlocal y_offset
+                td = target_data[n]
+                coord = SkyCoord(td['ra'], td['dec'], unit="rad")
+                x_pix, y_pix = skycoord_to_pixel(coord, wcs)
+                x_offset = int(x_pix - header_n['CRPIX1'])
+                y_offset = int(y_pix - header_n['CRPIX2'])
+                if abs(x_offset) > x_frac:
+                    x_offset = int(copysign(x_frac, x_offset))
+                if abs(y_offset) > y_frac:
+                    y_offset = int(copysign(y_frac, y_offset))
+                data_x_range = [x + x_offset for x in data_x_range]
+                data_y_range = [y + y_offset for y in data_y_range]
+                x_frac += x_offset
+                y_frac += y_offset
+            data = data[data_y_range[0]:data_y_range[1], data_x_range[0]:data_x_range[1]]
+            # Set new coordinates for Reference Pixel w/in smaller window
+            header_n['CRPIX1'] -= x_frac
+            header_n['CRPIX2'] -= y_frac
 
+        z_interval = ZScaleInterval().get_limits(data)  # set z-scale: responsible for vast majority of compute time
         plt.imshow(data, cmap='gray', vmin=z_interval[0], vmax=z_interval[1])
 
         # If first few frames, add 5" and 15" reticle
         if current_count < 6 and fr != init_fr or show_reticle:
-            if plot_source and header_n['CRPIX1'] > 0 and header_n['CRPIX2'] > 0:
+            if plot_source and (data.shape[1] > header_n['CRPIX1'] > 0) and (data.shape[0] > header_n['CRPIX2'] > 0):
```

Comment:
Is the order of the `data.shape` indices really opposite to the WCS indexes ? This isn't specific to rotated 0.4m data only is it ?

Reply:
The rotation here is handled by the wcs. This is purely a pixel coordinate issue.
The data array is defined as an array of rows, so the `y` length will be the number of rows, or top level of array dimensions, while the `x` length will be the size of each row, so the 2nd order dimensions. Which will be the case for every CCD.

In this case, it shouldn't matter because we are generally reshaping the field into a square, except for spectroscopy guide frames, and we don't plot a chip center for those.